### ISSUE TITLE
6.0 Update gRPC Client + Server app samples +tutorial

### ISSUE THREAD

**Issue by wadepickett** [dotnet/AspNetCore.Docs#23076]
Update the samples and tutorial to work on .NET SDK 6.0 and latest VS (Windows) 2022 preview and VS Code

See related issue #23243 for VS 2022 for Mac Preview update.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1d298a6c-8ddc-dbc8-d219-0fa7138442c2
* Version Independent ID: b96be346-958a-cfcd-d80a-c352d64223e8
* Content: [Create a .NET Core gRPC client and server in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/grpc/grpc-start?view=aspnetcore-5.0&tabs=visual-studio)
* Content Source: [aspnetcore/tutorials/grpc/grpc-start.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/grpc/grpc-start.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @JunTaoLuo
* Microsoft Alias: **johluo**

**Comment by johnseed** [comment]
Current sample does not working on VS2022, if start the greet server with VS2022, no matter Ctrl F5 or F5, the greet client will crash, however, if just double click greet server exe file, it works fine.
```bash
info: Microsoft.Hosting.Lifetime[0]
      Now listening on: https://localhost:5001
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Development
info: Microsoft.Hosting.Lifetime[0]
      Content root path: C:\Users\John\source\repos\AspNetCore.Docs\aspnetcore\tutorials\grpc\grpc-start\sample\GrpcGreeter
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/2 POST https://localhost:5001/greet.Greeter/SayHello application/grpc
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[0]
      Executing endpoint 'gRPC - /greet.Greeter/SayHello'
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[1]
      Executed endpoint 'gRPC - /greet.Greeter/SayHello'
fail: Microsoft.AspNetCore.Server.Kestrel[13]
      Connection id "0HMBELNUP840G", Request id "0HMBELNUP840G:00000001": An unhandled exception was thrown by the application.
System.ObjectDisposedException: Cannot write to the response body, the response has completed.
Object name: 'HttpResponseStream'.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponsePipeWriter.<ValidateState>g__ThrowObjectDisposedException|14_0()
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseStream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at Microsoft.WebTools.BrowserLink.Net.ScriptInjectionFilterStream.WriteToOutputStreamAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken& cancellationToken)
   at Microsoft.WebTools.BrowserLink.Net.ScriptInjectionFilterStream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at System.IO.Stream.WriteAsync(ReadOnlyMemory`1 buffer, CancellationToken cancellationToken)
   at System.IO.Pipelines.StreamPipeWriter.FlushAsyncInternal(CancellationToken cancellationToken)
   at System.IO.Pipelines.StreamPipeWriter.CompleteAsync(Exception exception)
   at Microsoft.AspNetCore.Http.StreamResponseBodyFeature.CompleteAsync()
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.FireOnCompletedAwaited(Task currentTask, Stack`1 onCompleted)
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished in 91.449ms 200 application/grpc
```

I upgrade the project to .net6, start greet server with VS2022, the error persists.
```bash
Grpc.Core.RpcException: 'Status(StatusCode="Internal", Detail="Failed to deserialize response message.")'
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Grpc.Net.Client.Internal.HttpClientCallInvoker.BlockingUnaryCall[TRequest,TResponse](Method`2 method, String host, CallOptions options, TRequest request)
   at Grpc.Core.Interceptors.InterceptingCallInvoker.<BlockingUnaryCall>b__3_0[TRequest,TResponse](TRequest req, ClientInterceptorContext`2 ctx)
   at Grpc.Core.ClientBase.ClientBaseConfiguration.ClientBaseConfigurationInterceptor.BlockingUnaryCall[TRequest,TResponse](TRequest request, ClientInterceptorContext`2 context, BlockingUnaryCallContinuation`2 continuation)
   at Grpc.Core.Interceptors.InterceptingCallInvoker.BlockingUnaryCall[TRequest,TResponse](Method`2 method, String host, CallOptions options, TRequest request)
   at Greet.Greeter.GreeterClient.SayHello(HelloRequest request, CallOptions options) in C:\Users\John\source\repos\GrpcDemo\GrpcGreeterClient\obj\Debug\net6.0\Protos\GreetGrpc.cs:line 119
   at Greet.Greeter.GreeterClient.SayHello(HelloRequest request, Metadata headers, Nullable`1 deadline, CancellationToken cancellationToken) in C:\Users\John\source\repos\GrpcDemo\GrpcGreeterClient\obj\Debug\net6.0\Protos\GreetGrpc.cs:line 108
   at <Program>$.<Main>$(String[] args) in C:\Users\John\source\repos\GrpcDemo\GrpcGreeterClient\Program.cs:line 18
```

**Comment by wadepickett** [comment]
Fixed and closed by #23253
